src/foam/blob/InputStreamBlob.java

```diff
@@ -6,28 +6,50 @@
 
 package foam.blob;
 
-import com.google.common.io.ByteStreams;
-
+import java.io.BufferedInputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 
 public class InputStreamBlob
     extends foam.blob.AbstractBlob
 {
-  protected byte[] buffer_;
+  protected long size_;
+  protected BufferedInputStream reader_;
 
-  public InputStreamBlob(java.io.InputStream in) throws IOException {
-    buffer_ = ByteStreams.toByteArray(in);
+  public InputStreamBlob(java.io.InputStream in, long size) throws IOException {
+    this.size_ = size;
+    this.reader_ = new BufferedInputStream(in);
   }
 
   @Override
   public Buffer read(Buffer buffer, long offset) {
-    int length = (int) (getSize() - offset);
-    return new Buffer(length, ByteBuffer.wrap(buffer_, (int) offset, length));
+    try {
+      int outOffset = 0;
```

Comment:
Since InputStream does not support seeking we should probably assert that offset is the next sequential offset after the previous read like we do for servletrequestblob.

Reply:
Good idea